### PR TITLE
Feature : Add deepks_v_delta, which can help to train DeepKS model with loss term about Hamiltonian, psi and band.

### LINKED ISSUE #2805: Enable Hamiltonian/psi/band training for DeePKS model
### Background

The current version of ABACUS can only support training DeePKS model on energy, force, stress, bandgap etc.
But it can't support training on Hamiltonian, as well as its eigenvalues (band) and eigenvectors (psi), which are important for electronic structure properties.

### Describe the solution you'd like

Enable Hamiltonian/psi/band training for DeePKS model

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xuan112358**
#2805

### CODE REVIEW

**Comment by wenfei-li** [comment]
Looks good to me.

File: source/module_io/input.h [hunk 1]
```diff
@@ -481,6 +481,7 @@ class Input
                             // 2022-1-12
     bool deepks_scf; //(need libnpy and libtorch) if set 1, a trained model would be needed to cal V_delta and F_delta
     bool deepks_bandgap; // for bandgap label. QO added 2021-12-15
+    int deepks_v_delta;//for v_delta label. xinyuan added 2023-2-15
```
Comment (dyzheng) [inline source/module_io/input.h:488]: Please add documentations for your new parameter.

File: source/module_hamilt_lcao/module_deepks/LCAO_deepks.h [hunk 2]
```diff
@@ -219,6 +228,10 @@ class LCAO_Deepks
     void init_orbital_pdm_shell(const int nks);
     void del_orbital_pdm_shell(const int nks);
 
+    //for v_delta label calculation; xinyuan added on 2023-2-22
```
Comment (dyzheng) [inline source/module_hamilt_lcao/module_deepks/LCAO_deepks.h:231]: Please add detail annotation for new functions and try to add UnitTests for them.

File: source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp [hunk 3]
```diff
@@ -362,4 +382,56 @@ void LCAO_Deepks::del_orbital_pdm_shell(const int nks)
     return;
 }
 
+void LCAO_Deepks::init_v_delta_pdm_shell(const int nks,const int nlocal)
```
Comment (hongriTianqi) [inline source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp:385]: Please add unit test for newly added functions.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp (hunk 3 above)
Comment (dyzheng) [reply source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp:385]: New integrate tests has covered the function.

File: source/module_io/input.h (hunk 1 above)
Comment (dyzheng) [reply source/module_io/input.h:488]: New integrate tests has covered the function.

**Review by dyzheng — APPROVED** [review]

File: source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp (hunk 3 above)
Comment (hongriTianqi) [reply source/module_hamilt_lcao/module_deepks/LCAO_deepks.cpp:385]: There are many newly added functions, they will reduce the coverage rate of unit tests.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

**Comment by mohanchen** [comment]
Will PR again in the near future

**Pull request closed without merging**